sat/cli/bmccreds/creds_manager.py

```diff
@@ -67,12 +67,15 @@ class BMCCredsManager:
     CHASSIS_BMC_XNAME_REGEX = re.compile(r'x\d+c\d+b\d+$')
     ROUTER_BMC_XNAME_REGEX = re.compile(r'x\d+c\d+r\d+b\d+$')
 
-    def __init__(self, password, xnames, domain, force, report_format):
+    def __init__(self, password, xnames, domain, force, report_format, length=RANDOM_PASSWORD_LENGTH,
+                 allowed_chars=VALID_CHAR_SETS):
         self.password = password
         self.xnames = [XName(xname) for xname in xnames]
         self.domain = domain
         self.force = force
         self.report_format = report_format
+        self.length = length
+        self.allowed_chars = allowed_chars
```

Comment:
These are not yet being passed in from the parsed command-line arguments. Those need to be passed in, and you will need to decide between `--password-chars` and `--password-char-sets` based on what was specified by the user.

Reply:
Decision has been taken in such a way that it checks `--password-chars` or `--password-char-sets` given by user if not use a default values i.e, `VALID_CHAR_SETS_STRING` which is a string of 'alpha,numeric,symbols'